packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -551,6 +567,15 @@ class SelectableRegionState extends State<SelectableRegion> with TextSelectionDe
           case TargetPlatform.macOS:
           case TargetPlatform.linux:
           case TargetPlatform.windows:
+            // It is impossible to extend the selection when the shift key is
+            // pressed and the start of the selection has not been initialized.
+            // In this case we fallback on collapsing the selection to first
+            // initialize the selection.
+            final bool isShiftPressedValid = _isShiftPressed && _selectionDelegate.value.startSelectionPoint != null;
```

Comment:
on mac chrome, if the existing selection is not collapsed, it looks like it will collapse the selection

Reply:
Trying this in macOS chrome but I do not see this behavior.

https://github.com/flutter/flutter/assets/948037/f358dd2c-4869-4408-ba54-f047448b4357

This PR is mainly trying to achieve the Linux/Windows/Android behavior though, which moves the selection end edge. The macOS is a bit more involved to implement because we have to determine what edge is closest to the tapped position (WRT to the text) and move that edge. I don't think that there's a straight forward way to do this right now, I tried to do this by comparing the global coordinates but that will not work since the edge that is moved is determined by text distance. I will attempt the macOS behavior in a later PR.